docs/tables/yml_key_value.md

```diff
@@ -0,0 +1,242 @@
+    size: 8
+    price: 133.7
+    quantity: 1
+bill-to: &id001
+street: |
+  123 Tornado Alley
+  Suite 16
+city: East Centerville
+state: KS
+ship-to: *id001
+specialDelivery: >
+  Follow the Yellow Brick
+  Road to the Emerald City.
+  Pay no attention to the
+  man behind the curtain.
+```
+
+and the query is:
+
+```sql
+with items as (
+  select
+    subpath(key_path, 0, 2) as item,
+    subpath(key_path, 2, 3) as data,
+    value
+  from
+    yml_key_value
+  where
+    path = '/Users/myuser/yml/invoice.yml'
+    and key_path ~ 'items.*'
+)
+select
+  max(case when data = 'part_no' then value else null end) as part_no,
+  max(case when data = 'description' then value else null end) as item_name,
+  max(case when data = 'size' then value else null end) as size,
+  max(case when data = 'quantity' then value else null end) as quantity,
+  max(case when data = 'price' then value else null end) as price
+from
+  items
+group by
+  item;
+```
+
+```sh
++---------+-----------------------------+--------+----------+-------+
+| part_no | item_name                   | size   | quantity | price |
++---------+-----------------------------+--------+----------+-------+
+| A4786   | Water Bucket (Filled)       | <null> | 4        | 1.47  |
+| E1628   | High Heeled "Ruby" Slippers | 8      | 1        | 133.7 |
++---------+-----------------------------+--------+----------+-------+
+```
+
+or, you can check the value for a particular key:
+
+```sql
+with items as (
+  select
+    subpath(key_path, 0, 2) as item,
+    subpath(key_path, 2, 3) as data,
+    value
+  from
+    yml_key_value
+  where
+    path = '/Users/myuser/yml/invoice.yml'
+    and key_path ~ 'items.*'
+),
+pivot_tables as (
+  select
+  max(case when data = 'part_no' then value else null end) as part_no,
+  max(case when data = 'description' then value else null end) as item_name,
+  max(case when data = 'size' then value else null end) as size,
+  max(case when data = 'quantity' then value else null end) as quantity,
+  max(case when data = 'price' then value else null end) as price
+from
+  items
+group by
+  item
+)
+select * from pivot_tables where part_no = 'E1628';
+```
+
+### Casting column data for analysis
+
+Text columns can be easily cast to other types:
+
+```sql
+with items as (
+  select
+    subpath(key_path, 0, 2) as item,
+    subpath(key_path, 2, 3) as data,
+    value
+  from
+    yml_key_value
+  where
+    path = '/Users/myuser/yml/invoice.yml'
+    and key_path ~ 'items.*'
+)
+select
+  max(case when data = 'part_no' then value else null end) as part_no,
+  max(case when data = 'description' then value else null end) as item_name,
+  (max(case when data = 'size' then value else null end))::integer as size,
+  (max(case when data = 'quantity' then value else null end))::integer as quantity,
+  (max(case when data = 'price' then value else null end))::float as price
+from
+  items
+group by
+  item;
+```
+
+### Searching key paths
```

Comment:
As suggested above, I'd move the example(s) using `subpath` down to this section and explain a bit more about it